query/graphql/src/main/java/org/codice/ddf/admin/graphql/transform/GraphQLTransformOutput.java

```diff
@@ -200,4 +203,39 @@ public String createOutputObjectFieldTypeName(String fieldTypeName) {
                 transformEnum.getEnumTypeProvider(),
                 outputTypeProvider);
     }
+
+    private String toPrettyString(FunctionField<DataType> funcField,
+            List<ErrorMessage> customMessage) {
+        String functionName = funcField.fieldName();
+        List<Object> args = funcField.getArguments()
+                .stream()
+                .map(Field::getValue)
+                .collect(Collectors.toList());
+
+        return Boon.toPrettyJson(hidePassword(toMap(functionName, args, customMessage)));
+    }
+
+    private Object hidePassword(Object result) {
```

Comment:
I tried cleaning up this method a bit. I think we should avoid typecasting to implementations.

```
    private static final List<String> BLACK_LIST_WORDS = ImmutableList.of("pass");
    private static final String HIDDEN_FLAG = "******";

    private Object blackList(Object result) {
        if (result instanceof List) {
            ((List) result).forEach(this::blackList);
        } else if (result instanceof Map) {
            for (Map.Entry<String, Object> entry : ((Map<String, Object>) result).entrySet()) {
                if(entry.getValue() instanceof Map || entry.getValue() instanceof List)  {
                    blackList(entry.getValue());
                } else {
                    boolean containsBlackListWord = BLACK_LIST_WORDS.stream()
                            .anyMatch(blackListWord -> blackListWord.toLowerCase()
                                    .contains(entry.getKey()
                                            .toLowerCase()));
                    if(containsBlackListWord) {
                        entry.setValue(HIDDEN_FLAG);
                    }
                }
            }
        }

        return result;
    }
```